packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -147,13 +147,26 @@ export interface FunctionProps {
    * What security group to associate with the Lambda's network interfaces.
    *
    * Only used if 'vpc' is supplied.
+   * Only use this or the securityGroupsList prop, not both at once
    *
    * @default - If the function is placed within a VPC and a security group is
-   * not specified, a dedicated security group will be created for this
-   * function.
+   * not specified, either by this or securityGroupsList prop, a dedicated security
+   * group will be created for this function.
    */
   readonly securityGroup?: ec2.ISecurityGroup;
 
+  /**
+   * What list of security groups to associate with the Lambda's network interfaces.
+   *
+   * Only used if 'vpc' is supplied.
+   * Only use this or the securityGroup prop, not both at once
+   *
+   * @default - If the function is placed within a VPC and a security group is
+   * not specified, either by this or securityGroup prop, a dedicated security
+   * group will be created for this function.
+   */
+  readonly securityGroupsList?: ec2.ISecurityGroup[];
```

Comment:
would name this `securityGroups` and deprecate `securityGroup` (`@deprecated`).

Reply:
Yes please

Reply:
Yeah, makes sense. I didn't call it `securityGroups` just because the difference in the props names will be so subtle that, it can confuse the users. But marking the older one as deprecated can help.